modules/components/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java

```diff
@@ -332,4 +332,22 @@ private void handleNull(Object obj, String msg) throws MLDataProcessingException
         }
     }
 
+    public void deleteDataset(int tenantId, String userName, long datasetId) throws MLDataProcessingException {
+        try {
+            databaseService.deleteDataset(datasetId);
+            log.info(String.format("[Deleted] [dataset] %s of [user] %s of [tenant] %s", datasetId, userName, tenantId));
+        } catch (DatabaseHandlerException e) {
+            throw new MLDataProcessingException(e);
```

Comment:
Wouldn't it better to specify a functional error message in the MLDataProcessingException rather than directly wrapping the DatabaseHandlerException in it?